fastly-edge-assignments/src/handlers/assignments.rs

```diff
@@ -0,0 +1,241 @@
+use eppo_core::configuration_store::ConfigurationStore;
+use eppo_core::eval::{Evaluator, EvaluatorConfig};
+use eppo_core::ufc::UniversalFlagConfig;
+use eppo_core::{Attributes, Configuration, SdkMetadata};
+use fastly::http::StatusCode;
+use fastly::kv_store::KVStoreError;
+use fastly::{Error, KVStore, Request, Response};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[derive(Debug, Deserialize)]
+struct RequestBody {
+    subject_key: String,
+    subject_attributes: Arc<Attributes>,
+    // TODO: Add bandit actions
+    // #[serde(rename = "banditActions")]
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // bandit_actions: Option<HashMap<String, serde_json::Value>>,
+}
+
+// Response
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "UPPERCASE")]
+enum AssignmentFormat {
+    Precomputed,
+}
+
+#[derive(Debug, Serialize)]
+struct Environment {
+    name: String,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct FlagAssignment {
+    allocation_key: String,
+    variation_key: String,
+    variation_type: String,
+    variation_value: serde_json::Value,
+    extra_logging: HashMap<String, serde_json::Value>,
+    do_log: bool,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct AssignmentsResponse {
+    created_at: i64,
+    format: AssignmentFormat,
+    environment: Environment,
+    flags: HashMap<String, FlagAssignment>,
+}
+
+const KV_STORE_NAME: &str = "edge-assignment-kv-store";
+const SDK_KEY_QUERY_PARAM: &str = "apiKey"; // For legacy reasons this is named `apiKey`
```

Comment:
minor: given this is a completely new service, I think we can go with `sdkKey` as there are no legacy clients?

Reply:
I want to do this as well - @sameerank used our existing javascript client to access this service to setup a connection for our eppo.cloud QA environment so we can start collect telemetry on real-world use. https://github.com/Eppo-exp/js-sdk-common/pull/137

I have a couple of goals to evolve the network architecture of this service:

* Serve it behind our existing CDN (with no caching) to be able to share the same domain name and SSL certificate
* Since Sameeran developed the new client separately from the old we have a lot of control over the query parameters and should make this change; or even better, pass the auth information in a `Authorization` header.